### PR TITLE
Upgrade stylelint config

### PR DESCRIPTION
**Pull request by adidahiya**
Pretty much the only meaningful change here is the removal of newlines between mixins and rules in declaration blocks, which I approve of 👍

### CODE REVIEW

File: packages/core/src/components/button/_button-group.scss [hunk 1]
```diff
@@ -147,7 +146,6 @@ Styleguide components.button-group.css
 
       &::after {
         @include pt-button-minimal-divider();
-
```
Comment (giladgray) [inline packages/core/src/components/button/_button-group.scss]: i've always found this whitespace helpful in selectors with lots of properties. 
is it now banned by stylelint?

Comment (adidahiya) [inline packages/core/src/components/button/_button-group.scss]: It is banned with the new `declaration-empty-line-before` rule and I couldn't figure out a clean way to enforce our own style (I want to enforce no blank lines between plain rule declarations). I don't think this whitespace is helpful; syntax highlighting of the `@ignore` token works pretty great IMO

Comment (llorca) [inline packages/core/src/components/button/_button-group.scss]: I'm personally a fan of that whitespace, has helped me parse the code easier more than once (I've even added that space in places where it wasn't before). Why _forbid_ it?

Comment (llorca) [inline packages/core/src/components/button/_button-group.scss]: I get what's forbidden, simple new lines before rule declaration. Maybe an other option would be to move those `@include`s at the bottom of all the rules? With a new line even maybe? Or do as you suggested, just remove that new line everywhere but I'm gonna miss it 😢

**Pull request merged**